High Cardinality Source Schema Export (Public Beta) (#4346)
segmentio/segment-docs: src/connections/destination-data-control.md

![A screenshot of the source schema page. The user is hovered over the integrations column, and a popup showing the three integrations connected to the event appears.](images/destination-control.png)

## How do I block or disable specific events and properties from being sent to all Destinations?

If you no longer want to track an event, you can either remove it from your code or, if you're on the Business plan, you can block track calls right from the Segment UI on your Source Schema page by adjusting the toggle for each event.

![A screenshot of the source schema page. The toggles in the integrations column are all enabled.](/docs/protocols/images/event-filters.png)

Once you block an event in Segment, Segment stops forwarding it to all of your Destinations, including your warehouses. You can remove it from your code at your leisure. In addition to blocking track calls, Business plan customers can block all Page and Screen calls, as well as Identify traits and Group properties. 

## Export your Source Schema

> Segment allows users with Source View permissions to download Source Schemas as a CSV file, maximizing portability and access to event data. You can download a copy of your schema by visiting the Source Schema page. 

Reviewer: `permission`?

Author: I went back and forth on this one: I settled on permissions because as with source read-only, you had view access to multiple things (ie permission to view multiple things.) I'll ask the docs team/update the role to say Source Read-only so it matches the UI!

Author: The docs team agreed on `permissions`, so I'll stick with that for now!

> Segment allows users with Source View permissions to download Source Schemas as a CSV file, maximizing portability and access to event data. You can download a copy of your schema by visiting the Source Schema page. 

Author: ```suggestion
Segment allows users with Source Read-only permissions to download Source Schemas as a CSV file, maximizing portability and access to event data. You can download a copy of your schema by visiting the Source Schema page. 
```

> success ""
> You can export Track, Identify, and Group Source Schemas.

### Download a CSV

To download a Source Schema CSV file: 
1. Sign into Segment and select a source.
2. Click the **Schema** tab in the source header.

> 3. On the Source Schema page, select an event type (Track, Identify, or Group)

Reviewer 2: should this be `event type` or `schema type`? We currently use `schema type` in the download history table

Author: I'll go with `schema type` if that's what's in the download history table!

6. Once the file status column indicates that the download was successful, click the **Download CSV** link to download your CSV to your computer.<br/> The Source Schema CSV name has the following format:<br/>`workspace-slug-source-slug-schema-type-yyyy-mm-dd-hh-mm`

> note "All events and properties are now included in the CSV file"
> When you exporting the Source Schema, all events and properties are included in the CSV file regardless of the filters or search parameters currently applied to the Source Schema view. 

### View download history

You can view the last 14 days' worth of Source Schema exports on the Download History page. 

To access the Download History page:
1. Sign into Segment and select a source.
2. Click the **Schema** tab in the source header.
3. Click the **View Download History** link.

### Track event CSV format
The Track event CSV file contains the following columns:
- Event Name
- Last Seen At (UTC)
  - If greater than 30 days, the value is "more than 30 days ago"
- Property Name
- Allowed
- Blocked
- Total

> - Planned (available for Protocols customers with a connected Tracking Plan)

Reviewer 2: We also include labels created by the user

Author: I can't find a CSV in the test source with a labels column - is the title of the column just `Label`?

  - Values are "planned" or "unplanned"

### Identity and Group event CSV format
The Identify and Group CSV file contains the following columns:
- Trait Name
- Last Seen At (UTC)
  - If greater than 30 days, the value is "more than 30 days ago"
- Allowed
- Blocked
- Total
- Planned (available for Protocols customers with a connected Tracking Plan)
  - Values are "planned" or "unplanned"

> 

Reviewer 2: Same here about including labels

Segment allows users with Source Read-only permissions to download Source Schemas as a CSV file, maximizing portability and access to event data. You can download a copy of your schema by visiting the Source Schema page. 

> success ""
> You can export Track, Identify, and Group Source Schemas.

### Download a CSV

To download a Source Schema CSV file: 
1. Sign into Segment and select a source.
2. Click the **Schema** tab in the source header.
3. On the Source Schema page, select a schema type (Track, Identify, or Group)
4. Click the **Download CSV** button. <br/> A toast pops up on the top of the page, with the message *"Your file is processing. When your file is ready it will be available to download from the Download History page."* 
5. Open the Download History page by clicking the link in the toast or following the instructions in the [view download history](#view-download-history) section.
6. Once the file status column indicates that the download was successful, click the **Download CSV** link to download your CSV to your computer.<br/> The Source Schema CSV name has the following format:<br/>`workspace-slug-source-slug-schema-type-yyyy-mm-dd-hh-mm`

> > note "All events and properties are now included in the CSV file"

Author: @mschasz - does this note adequately cover the callout about all data being included in the exported csv, or should we add a note about filtering the data after a user downloads it?

Reviewer 2: I think this should suffice!

> > When you exporting the Source Schema, all events and properties are included in the CSV file regardless of the filters or search parameters currently applied to the Source Schema view. 

Author: ```suggestion
> When you export the Source Schema, all events and properties are included in the CSV file regardless of the filters or search parameters currently applied to the Source Schema view. 
```

## How do I block or disable specific events and properties from being sent to all Destinations?

If you no longer want to track an event, you can either remove it from your code or, if you're on the Business plan, you can block track calls right from the Segment UI on your Source Schema page by adjusting the toggle for each event.

![A screenshot of the source schema page. The toggles in the integrations column are all enabled.](/docs/protocols/images/event-filters.png)

Once you block an event in Segment, Segment stops forwarding it to all of your Destinations, including your warehouses. You can remove it from your code at your leisure. In addition to blocking track calls, Business plan customers can block all Page and Screen calls, as well as Identify traits and Group properties. 

## Export your Source Schema

Segment allows users with Source Read-only permissions to download Source Schemas as a CSV file, maximizing portability and access to event data. You can download a copy of your schema by visiting the Source Schema page.

> success ""
> You can export Track, Identify, and Group Source Schemas.

### Download a CSV

> You can only download one Source Schema CSV per source and per schema type (Track, Identify, or Group) at the same time.

Reviewer 2: maybe we can word this per schema type (Track, Identify, or Group) per source?

> You can only download one Source Schema CSV per source and per schema type (Track, Identify, or Group) at the same time.

Author: ```suggestion
You can only download one Source Schema CSV schema type (Track, Identify, or Group) per source at the same time.
```

To download a Source Schema CSV file: 
1. Sign into Segment and select a source.
2. Click the **Schema** tab in the source header.

> 3. On the Source Schema page, select a schema type (Track, Identify, or Group)

Reviewer 2: we should also mention that they should select a timeframe (only options are the past 7 days or 30 days)

> 3. On the Source Schema page, select a schema type (Track, Identify, or Group)

Author: ```suggestion
3. On the Source Schema page, select a schema type (Track, Identify, or Group) and a timeframe (7 days or 30 days.)
```

4. Click the **Download CSV** button. <br/> A toast pops up on the top of the page, with the message *"Your file is processing. When your file is ready it will be available to download from the Download History page."* 
5. Open the Download History page by clicking the link in the toast or following the instructions in the [view download history](#view-download-history) section.

> 6. Once the file status column indicates that the download was successful, click the **Download CSV** link to download your CSV to your computer.<br/> The Source Schema CSV name has the following format:<br/>`workspace-slug-source-slug-schema-type-yyyy-mm-dd-hh-mm`

Author: ```suggestion
6. Once the file status column indicates that the download was successful, click the **Download CSV** link to download your CSV to your computer.<br/> The Source Schema CSV name has the following format:<br/>`workspaceSlug-sourceSlug-schemaType--yyyy-mm-dd--hh-mm-utc`
```

> 6. Once the file status column indicates that the download was successful, click the **Download CSV** link to download your CSV to your computer.<br/> The Source Schema CSV name has the following format:<br/>`workspaceSlug-sourceSlug-schemaType--yyyy-mm-dd--hh-mm-utc`

Reviewer 2: Can we add a note about retrying the download if there's an error?

> 6. Once the file status column indicates that the download was successful, click the **Download CSV** link to download your CSV to your computer.<br/> The Source Schema CSV name has the following format:<br/>`workspaceSlug-sourceSlug-schemaType--yyyy-mm-dd--hh-mm-utc`

Author: ```suggestion
6. Once the file status column indicates that the download was successful, click the **Download CSV** link to download your CSV to your computer. If the file status column shows that the download has failed, return to the Source Schema page and try the download again.<br/> The Source Schema CSV name has the following format:<br/>`workspaceSlug-sourceSlug-schemaType--yyyy-mm-dd--hh-mm-utc`
```

Author: @mschasz like this?

> note "All events and properties are now included in the CSV file"
> When you export a Source Schema, all events and properties are included in the CSV file regardless of the filters or search parameters currently applied to the Source Schema view. 

### View download history

You can view the last 14 days' worth of Source Schema exports on the Download History page. 

To access the Download History page:
1. Sign into Segment and select a source.
2. Click the **Schema** tab in the source header.
3. Click the **View Download History** link.

### Track event CSV format
The Track event CSV file contains the following columns:
- Event Name
- Last Seen At (UTC)

>   - If greater than 30 days, the value is "more than 30 days ago"

Reviewer 2: We should adjust this to include the other timeframe option of the past 7 days too

>   - If greater than 30 days, the value is "more than 30 days ago"

Author: ```suggestion
  - If greater than your selected timeframe (7 days or 30 days,) the value is "more than 7 days ago" or "more than 30 days ago"
```

- Property Name
- Allowed
- Blocked
- Total
- Planned (available for Protocols customers with a connected Tracking Plan)
  - Values are "planned" or "unplanned"

> note "Labels in your exported CSV"
> If you use [labels](/docs/protocols/tracking-plan/create/#add-a-label), they appear as columns in your CSV. The column headers are keys, and the column data contains values.

### Identity and Group event CSV format
The Identify and Group CSV files contain the following columns:
- Trait Name
- Last Seen At (UTC)

>   - If greater than 30 days, the value is "more than 30 days ago"

Reviewer 2: Same here - we should adjust this to include the other timeframe option of the past 7 days too

Author: So qq - if a user selects a timeframe of 7 days, and had an event that happened 8 days earlier, the "last seen at" column would show "more than 7 days ago"? @mschasz

>   - If greater than 30 days, the value is "more than 30 days ago"

Author: ```suggestion
  - If greater than the user's selected timeframe (7 days or 30 days,) the value is "more than 7 days ago" or "more than 30 days ago"
```
